sql/information_schema/information_schema.go

```diff
@@ -1355,6 +1429,49 @@ func (pit *informationSchemaPartitionIter) Close(_ *Context) error {
 	return nil
 }
 
+func (t *routineTable) AssignCatalog(cat Catalog) Table {
+	t.catalog = cat
+	return t
+}
+
+func (r *routineTable) AssignRoutines(p []*plan.Procedure) Table {
+	// TODO: should also assign functions
+	r.procedures = p
+	return r
+}
+
+// Name implements the sql.Table interface.
+func (r *routineTable) Name() string {
+	return r.name
+}
+
+// Schema implements the sql.Table interface.
+func (r *routineTable) Schema() Schema {
+	return r.schema
+}
+
+func (r *routineTable) String() string {
+	return printTable(r.Name(), r.Schema())
+}
+
+func (r *routineTable) Partitions(context *Context) (PartitionIter, error) {
```

Comment:
Subtype the information schema table so you don't have to reimplement these methods. This starts by moving the RoutineTable definition